src/interpreter.c

```diff
@@ -23,9 +23,38 @@ typedef struct {
     size_t ip; // Leak the currently-evaluating statement index to backtrace capture
     int preevaluation; // use special rules for pre-evaluating expressions (deprecated--only for ccall handling)
     int continue_at; // statement index to jump to after leaving exception handler (0 if none)
+    void *associated_frame_pointer;
+    jl_gcframe_t frame;
 } interpreter_state;
 
-#include "interpreter-stacktrace.c"
+#ifdef __clang_analyzer__
+
+extern void JL_GC_ENABLEFRAME(interpreter_state*) JL_NOTSAFEPOINT;
+
+// This is necessary, because otherwise the analyzer considers this undefined
+// behavior and terminates the exploration
+#define JL_GC_PUSHFRAME(frame, n)     \
+  frame = (interpreter_state*)alloca(sizeof(*frame) + sizeof(void*) * (n)); \
+  memset(&frame[1], 0, sizeof(void*) * n); \
+  _JL_GC_PUSHARGS((jl_value_t**)&frame[1], n);
+
+#else
+
+#define JL_GC_PUSHFRAME(frame,n)                                                    \
+  frame = (interpreter_state*)alloca(sizeof(*frame)+((n)*sizeof(jl_value_t*)));     \
+  ((void**)&frame[1])[-3] = NULL;                                                   \
```

Comment:
These negative indices are rather confusing. I can see this makes the gcframe layout manifestly compatible with the other places gc frames are created (rather than setting `frame->associated_frame_pointer` etc directly). Took me a bit of squinting to see that though.

Reply:
Right, it's not dealing with an interpreter_state, it's dealing with pushing a general gc-frame just like any other. I just happened to only implement `interpreter_state` support.